src/router/index.js

```diff
@@ -115,55 +120,55 @@ router.beforeEach((to, from, next) => {
   }
 
   if (to.matched.some((record) => record.meta.requiresAuth)) {
-    if (store.getters["auth/isAuthenticated"]) {
-      next();
+    // app has to be authenticated using third party auth if all the query params
+    // match the keys in `requiredAuthKeys` and they're not empty or undefined
+    var isThirdPartyAuth =
+      requiredAuthKeys.every((key) => Object.keys(to.query).includes(key)) &&
```

Comment:
Object.keys(to.query) -> can be saved in a variable? - being estimated every time for each key